src/OrchardCore.Modules/OrchardCore.Apis.GraphQL/GraphQLSettings.cs

```diff
@@ -9,8 +12,20 @@ public class GraphQLSettings
         public Func<HttpContext, object> BuildUserContext { get; set; }
 
         public bool ExposeExceptions { get; set; } = false;
+
         public int? MaxDepth { get; set; }
         public int? MaxComplexity { get; set; }
         public double? FieldImpact { get; set; }
+        public int MaxNumberOfResults { get; set; }
+        public MaxNumberOfResultsValidationMode MaxNumberOfResultsValidationMode { get; set; }
+
+        public IEnumerable<IValidationRule> ValidationRules { get; set; } = Enumerable.Empty<IValidationRule>();
+    }
+
+    public enum MaxNumberOfResultsValidationMode
+    {
+        Environment,
+        Debug,
```

Comment:
When I mentioned `debug` and `release` it was only as a reference to the bundling feature. I would call these `Disabled` and `Enabled` here. And maybe `Environment` could become `Default`, or `Automatic`.

Reply:
i know! i meant to change them hah! was using as placeholders while i thought of better ones :) which you've just done!